sdk/eventhub/azure-eventhubs-aio-partitionmanager-blobstorage/azure/eventhub/aio/partitionmanager/blobstorage/blobstoragepm.py

```diff
@@ -0,0 +1,125 @@
+            owner_id = ownership["owner_id"]
+
+            async with self._cached_ownership_locks[partition_id]:
+                metadata = {"owner_id": ownership["owner_id"]}
+                if "offset" in ownership:
+                    metadata["offset"] = ownership["offset"]
+                if "sequence_number" in ownership:
+                    metadata["sequence_number"] = ownership["sequence_number"]
+
+                etag = ownership.get("etag")
+                if etag:
+                    etag_match = {"if_match": '"'+etag+'"'}
+                else:
+                    etag_match = {"if_none_match": '"*"'}
+                try:
+                    blob_client = await self._container_client.upload_blob(
+                        name=partition_id, data=UPLOAD_DATA, overwrite=True, metadata=metadata, **etag_match
+                    )
+                    uploaded_blob_properties = await blob_client.get_blob_properties()
+                    ownership["etag"] = uploaded_blob_properties.etag
+                    ownership["last_modified_time"] = uploaded_blob_properties.last_modified.timestamp()
+                    self._cached_ownership_dict[partition_id] = ownership
+                except (ResourceModifiedError, ResourceExistsError):
+                    logger.info(
+                        "EventProcessor instance %r of eventhub %r consumer group %r lost ownership to partition %r",
+                        owner_id, eventhub_name, consumer_group_name, partition_id)
+                except Exception as err:
+                    logger.exception("An exception occurred when EventProcessor instance %r claim_ownership for "
+                                     "eventhub %r consumer group %r partition %r. The ownership is now lost",
+                                     owner_id, eventhub_name, consumer_group_name, partition_id, exc_info=err)
+                else:
+                    result.append(ownership)
+        return result
+
+    async def update_checkpoint(self, eventhub_name, consumer_group_name, partition_id, owner_id,
+                                offset, sequence_number) -> None:
+
+        metadata = {
+            "owner_id": owner_id,
+            "offset": offset,
+            "sequence_number": str(sequence_number)
+        }
+        async with self._cached_ownership_locks[partition_id]:
+            try:
+                blob_client = await self._container_client.upload_blob(
+                    name=partition_id, data=UPLOAD_DATA, metadata=metadata, overwrite=True)
+                uploaded_blob_properties = await blob_client.get_blob_properties()
+                cached_ownership = self._cached_ownership_dict[partition_id]
+                cached_ownership["etag"] = uploaded_blob_properties.etag
+                cached_ownership["last_modified_time"] = uploaded_blob_properties.last_modified.timestamp()
+            except (ResourceModifiedError, ResourceExistsError):
```

Comment:
Under what circumstances would you expect to get either of these exceptions?

Reply:
When multiple EventProcessors are running. The etag could be changed by another EventProcessor when an EventProcessor tries to upload blob.